subsys/bluetooth/host/hci_core.c

```diff
@@ -5921,13 +5920,32 @@ int bt_le_adv_start_internal(const struct bt_le_adv_param *param,
 		if (err) {
 			return err;
 		}
+
+		if (IS_ENABLED(CONFIG_BT_PERIPHERAL) &&
+		    param->options & BT_LE_ADV_OPT_CONNECTABLE) {
+			conn = bt_conn_add_le(param->id, BT_ADDR_LE_NONE);
+			if (!conn) {
+				return -ENOMEM;
+			}
+
+			bt_conn_set_state(conn, BT_CONN_CONNECT_ADV);
+		}
 	}
 
 	err = set_advertise_enable(true);
 	if (err) {
+		BT_ERR("Failed to start advertiser");
+		if (IS_ENABLED(CONFIG_BT_PERIPHERAL) && conn) {
```

Comment:
Since you're pre-initializing conn to NULL and only setting it to non-NULL in a branch that uses `IS_ENABLED(CONFIG_BT_PERIPHERAL)` I *think* you can drop the `IS_ENABLED()` here and the compiler should be smart enough to compile away the branch either way. It'll also make the code look safer since it won't raise the question "wait, are we leaking conn here if PERIPHERAL is not set but conn is set?".

Reply:
Looks like you are right. mesh_demo passed. Will check if shippable finds other configurations